### PR TITLE
Add new opt TrustProxy and support proxy request

### PR DESCRIPTION
**Pull request by liut**
Fixes to depend `github.com/tomasen/realip`

### CODE REVIEW

File: ipfilter.go [hunk 1]
```diff
@@ -400,11 +403,14 @@ type ipFilterMiddleware struct {
...
-	//use remote addr as it cant be spoofed
-	//TODO also check X-Fowarded-For and friends
-	ip, _, _ := net.SplitHostPort(r.RemoteAddr)
-	//show simple forbidden text
-	if !m.IPFilter.Allowed(ip) {
+	var remoteIP string
+	if m.opts.TrustProxy {
+		remoteIP = realip.FromRequest(r)
+	} else {
+		remoteIP, _, _ := net.SplitHostPort(r.RemoteAddr)
```
Comment (jpillora) [inline ipfilter.go]: `:=` should be `=`

Comment (liut) [reply ipfilter.go]: Yeah! sorry.

**Comment by jpillora** [comment]
Thanks!

**Comment by liut** [comment]
😄

**Pull request merged**